src/browser/api_protocol/api_handlers/application.js

```diff
@@ -350,71 +340,58 @@ function deregisterExternalWindow(identity, message, ack) {
 
 function externalWindowAction(identity, message, ack) {
     /* jshint bitwise: false */
-    let payload = message.payload;
-    const { uuid, name } = payload;
-
-    const SWP_HIDEWINDOW = 128;
-    const SWP_SHOWWINDOW = 64;
-    const SC_MAXIMIZE = 61488;
-    const SC_MINIMIZE = 61472;
-    const SC_RESTORE = 61728;
+    const { payload, payload: { type, uuid, name } } = message;
 
-    switch (payload.type) {
-        case 2:
-            // WM_DESTROY
+    switch (type) {
+        case WindowsMessages.WM_DESTROY:
             ofEvents.emit(route.externalWindow('close', uuid, name));
             break;
-        case 7:
-            // WM_SETFOCUS
+        case WindowsMessages.WM_SETFOCUS:
             ofEvents.emit(route.externalWindow('focus', uuid, name));
             break;
-        case 8:
-            // WM__KILLFOCUS
+        case WindowsMessages.WM__KILLFOCUS:
```

Comment:
This is undefined... `WM__KILLFOCUS` vs `WM_KILLFOCUS`. How can we confirm these?

Reply:
we could do WindowMessages.get('KILLFOCUS') which could log an error (or worse) if KILLFOCUS is unknown, but would be unusual coding and probably overkill.

Reply:
I searched for `__` inside a word and the above was the only hit:
```bash
egrep -r '\w+__\w+' src
```